guppylang/checker/errors/type_errors.py

```diff
@@ -202,7 +203,43 @@ def rendered_span_label(self) -> str:
             msg = "Unexpected assignment " + ("targets" if diff < -1 else "target")
         else:
             msg = "Not enough assignment targets"
-        return f"{msg} (expected {self.expected}, got {self.actual})"
+        at_least = "at least " if self.at_least else ""
+        return f"{msg} (expected {self.expected}, got {at_least}{self.actual})"
```

Comment:
The `at_least` message could probably go inside the `diff < 0` branch for clarity, as it should always be `false` otherwise.